src/main/java/com/sourcegraph/webhook/EventSerializer.java

```diff
@@ -0,0 +1,103 @@
+package com.sourcegraph.webhook;
+
+import com.atlassian.bitbucket.event.ApplicationEvent;
+import com.atlassian.bitbucket.event.pull.*;
+import com.atlassian.bitbucket.json.JsonRenderer;
+import com.atlassian.plugin.spring.scanner.annotation.imports.ComponentImport;
+import com.google.gson.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.text.SimpleDateFormat;
+import java.util.HashMap;
+import java.util.Map;
+
+@Component
+public class EventSerializer {
+    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm'Z'");
+    private static Map<Class<?>, Adapter> adapters = new HashMap<>();
+    private static JsonRenderer renderer;
+
+    private JsonObject payload;
+
+    private static JsonElement render(Object o) {
+        String raw = renderer.render(o, new HashMap<>());
```

Comment:
Java noob question: What is a `HashMap` doing here? Aren't we rendering an Object directly to a JSON String?

Reply:
The `HashMap` is for the `options` parameter of the `render` function. I think passing in a `null` will throw a null pointer exception, so I just pass in an empty one. 

I'm not exactly sure what the options do since the docs don't say much 🤔https://docs.atlassian.com/bitbucket-server/javadoc/4.0.2/spi/reference/com/atlassian/bitbucket/json/JsonRenderer.html

Reply:
I had a similar question. Can you assign this to a variable so it is clearer?

```suggestion
        HashMap<> options = new HashMap<>(); 
        String raw = renderer.render(o, options);
```

Do we really have to indirect through a JSON string? There is no method to go directly from a regular object to a JsonElement?

Reply:
The bitbucket API only exposes that single method `render` that converts bitbucket objects (PullRequest, User, Ref, etc) to a JSON string. The other approach would be manually serializing these objects like how events are currently being handled to avoid this redundant operation.